intrinsicsolver/fixFunctions.go

```diff
@@ -63,6 +81,28 @@ func FixFunctions(template []byte, logger *logger.Logger) ([]byte, error) {
 	return preprocessedTemplate, nil
 }
 
+// Expands the function name to it's long form without a colon. For example - Fn::FindInMap.
+func longForm(name string) string {
+	var fullName string
+	if name != "Ref" {
+		fullName = "Fn::" + name
```

Comment:
What if the name contains already `Fn::`. Will it happen?

Reply:
Those functions are used inside `elongateForms` which is scanning the file for short names. It has a condition if there are long names already - it simply surrounds it with quotes.